src/python/pants/backend/python/tasks/resolve_requirements_task_base.py

```diff
@@ -34,6 +35,12 @@ def prepare(cls, options, round_manager):
     round_manager.require_data(PythonInterpreter)
     round_manager.optional_product(PythonRequirementLibrary)  # For local dists.
 
+  @staticmethod
+  def tgt_closure_has_native_sources():
+    local_dist_tgts = self.context.targets(is_local_python_dist)
+    if local_dist_tgts:
```

Comment:
Should be able to remove the `if` here: `def targets` should always return a collection. But more generally: this method should always result in an explicit return, and dropping the `if` solves that.